test/document_stores/test_pinecone.py

```diff
@@ -109,102 +112,157 @@ def docs_all_formats(self) -> List[Union[Document, Dict[str, Any]]]:
                     "date": "2019-01-01",
                     "numeric_field": 0.0,
                     "odd_document": True,
+                    "year": "2020",
+                },
+            ),
+            Document(
+                content="My name is Adele and I live in London",
+                meta={
+                    "meta_field": "test-5",
+                    "name": "file_5.txt",
+                    "date": "2019-01-01",
+                    "numeric_field": 0.0,
+                    "odd_document": True,
+                    "year": "2021",
                 },
             ),
             # Without meta
             Document(content="My name is Ahmed and I live in Cairo"),
+            Document(content="My name is Bruce and I live in Gotham"),
+            Document(content="My name is Peter and I live in Quahog"),
         ]
 
     @pytest.fixture
-    def docs(self, docs_all_formats: List[Union[Document, Dict[str, Any]]]) -> List[Document]:
+    def documents(self, docs_all_formats: List[Union[Document, Dict[str, Any]]]) -> List[Document]:
         return [Document.from_dict(doc) if isinstance(doc, dict) else doc for doc in docs_all_formats]
 
     #
     #  Tests
     #
 
-    @pytest.mark.pinecone
+    @pytest.mark.integration
+    def test_ne_filters(self, ds, documents):
+        ds.write_documents(documents)
+
+        result = ds.get_all_documents(filters={"year": {"$ne": "2020"}})
+        assert len(result) == 3
+
+    @pytest.mark.integration
+    def test_get_label_count(self, ds, labels):
+        with pytest.raises(NotImplementedError):
+            ds.get_label_count()
+
+    # NOTE: the SQLDocumentStore behaves differently to the others when filters are applied.
```

Comment:
This is from the SQLDocumentStore but the comment here does not need to mention it in my opinion.